### ISSUE TITLE
[DMG] HALT should have no effect with disabled interrupts

### ISSUE THREAD

**Issue by NasenSpray** [shonumi/gbe-plus#27]
```
DI
HALT //no effect, continue as normal
```

**Comment by shonumi** [comment]
The effect of HALT is actually a bit more complicated than simply continuing as normal (well, on anything that's not a GBC). The following instruction is "skipped", or rather its opcode gets executed twice. It's unusual behavior, but only for DMG, MBP, and SGB units. It's not a one-line fix, so I'll get to this later (long work day today...)

**Comment by NasenSpray** [comment]
yeah, I've read about that. In my fork, I only fixed the not HALTing part:

```
    //HALT
  case 0x76:
    halt = interrupt;
    cycles += 4;
    break;
```

**Comment by shonumi** [comment]
Forgot to close this, but it should be fixed as of 37318afadc95de83d506c9e2f56ca72c698b2c8a
